Documentation/contributing/development/dev_setup.rst

```diff
@@ -579,12 +579,34 @@ Debugging
 Datapath code
 ^^^^^^^^^^^^^
 
-The tool ``cilium monitor`` can also be used to retrieve debugging information
-from the eBPF based datapath. Debugging messages are sent if either the
-``cilium-agent`` itself or the respective endpoint is in debug mode. The debug
-mode of the agent can be enabled by starting ``cilium-agent`` with the option
-``--debug`` enabled or by running ``cilium config debug=true`` for an already
-running agent. Debugging of an individual endpoint can be enabled by running
+The image below shows the options that could be used as startup options by
+``cilium-agent`` (see upper blue box) or could be changed at runtime by running
+``cilium config <option(s)>`` for an already running agent (see lower blue box).
+For instance, if you want to enable the debug mode (including the load
+balancer), you may start ``cilium-agent`` with ``--debug-verbose=datapath`` or
+run ``cilium config debug=true debugLB=true`` for an already running agent. For
+each option, there is a helper function which can be used by the developer to
+print debugging information in the tool ``cilium monitor``: ``cilium_dbg()`` for
+``DEBUG`` and ``cilium_dbg_lb()`` for ``DEBUG_LB``. 
```

Comment:
Nice work! If I may, I think the beginning of this paragraph is a bit abrupt. Let's keep in mind that people coming to read that section are looking for a way to enable debug logs, because they likely have issues in their datapath: The first thing we give them is a big diagram with plenty of possibilities.

Let's try to bring it in a more gentle way, maybe something along:
```
The tool ``cilium monitor`` can also be used to retrieve debugging information
from the eBPF based datapath. It relies on ``cilium_dbg()`` and a few other related
function in the code. These functions are inactive by default and must be enabled
by passing the relevant options to the ``cilium-agent`` or by enabling it manually
from a running agent. The image below shows the details of ...
```
[Then give some info on the commands, as you do already.]

Another possibility could even be to provide the commands that enable the highest level of debugging first, and to explain things after that. Thinking of:

```
The tool ``cilium monitor`` can also be used to retrieve debugging information
from the eBPF based datapath. To enable all log messages:
- Start the ``cilium-agent`` with `--debug-verbose=datapath``, or
- Run ``cilium config debug=true debugLB=true`` from an already running agent.

These options enable logging functions in the datapath: `cilium_dbg()`, etc. ...
The image below shows...
```

What do you think?

Reply:
Thank you for your suggestions! Absolutely!